packages/edit-site/src/components/block-editor/block-editor-provider/test/use-page-content-blocks.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { renderHook } from '@testing-library/react';
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import usePageContentBlocks from '../use-page-content-blocks';
+
+jest.mock( '@wordpress/blocks', () => {
+	return {
+		__esModule: true,
+		...jest.requireActual( '@wordpress/blocks' ),
+		createBlock( name, attributes = {}, innerBlocks = [] ) {
+			return {
+				name,
+				attributes,
+				innerBlocks,
+			};
+		},
+	};
+} );
+
+describe( 'usePageContentBlocks', () => {
+	const blocksList = [
+		createBlock( 'core/group', {}, [
+			createBlock( 'core/post-title' ),
+			createBlock( 'core/post-featured-image' ),
+			createBlock( 'core/query' ),
+			createBlock( 'core/post-content' ),
+		] ),
+		createBlock( 'core/query' ),
```

Comment:
Should we add a test were there are blocks as children of `core/query` to check that it doesn't iterate into children of that block type?

Reply:
This is a great idea. Will do!